### PR TITLE
authz: silo endpoints

### PR DESCRIPTION
**Pull request by davepacheco**
This adds authz protection to the silo public API endpoints.

### CODE REVIEW

File: nexus/src/db/datastore.rs [hunk 1]
```diff
@@ -31,7 +31,29 @@ use crate::authz::ApiResourceError;
 use crate::context::OpContext;
 use crate::db::fixed_data::role_assignment_builtin::BUILTIN_ROLE_ASSIGNMENTS;
 use crate::db::fixed_data::role_builtin::BUILTIN_ROLES;
-use crate::db::fixed_data::silo_builtin::SILO_ID;
+use crate::db::fixed_data::silo::{DEFAULT_SILO, SILO_ID};
+use crate::db::lookup::LookupPath;
+use crate::db::{
+    self,
+    error::{public_error_from_diesel_pool, ErrorHandler, TransactionError},
+    model::{
```
Comment (plotnick) [inline nexus/src/db/datastore.rs:39]: Should we just use `model::*` here?

**Review by plotnick — APPROVED** [review]
authz all the things!

File: nexus/src/db/datastore.rs (hunk 1 above)
Comment (david-crespo) [reply nexus/src/db/datastore.rs:39]: In other files I like importing `model` and then using the models qualified with `model::`. However in this file there isn't much opportunity for confusion.

Comment (davepacheco) [reply nexus/src/db/datastore.rs:39]: I don't have much opinion.  I'd be happy to use `model::` here as well.  But I'm not going to do it in this change -- it'd be a lot of extra delta.

**Pull request merged**